### PR TITLE
update setup guide

### PR DESCRIPTION
**Pull request by chendave**
- `usage.md` is mostly overlapping with `setup.md`
- `usage.md` has some content left behind
- there is no need to maintain two guides for the installation.

Signed-off-by: Dave Chen <dave.chen@arm.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by chendave** [comment]
cc @kevin-wangzefeng  @fisherxu  @kadisi  @subpathdev

File: docs/setup/setup.md [hunk 1]
```diff
@@ -21,6 +21,18 @@ KubeEdge is composed  of cloud and edge parts. It is built upon Kubernetes and p
...
+### Configuring MQTT mode
+
+The Edge part of KubeEdge uses MQTT for communication between deviceTwin and devices. KubeEdge supports 3 MQTT modes:
+1) internalMqttMode: internal mqtt broker is enabled.
+2) bothMqttMode: internal as well as external broker are enabled.
+3) externalMqttMode: only external broker is enabled.
+
+Use mode field in [edge.yaml](https://github.com/kubeedge/kubeedge/blob/master/edge/conf/edge.yaml#L4) to select the desired mode.
+
+To use KubeEdge in double mqtt or external mode, you need to make sure that [mosquitto](https://mosquitto.org/) or [emqx edge](https://www.emqx.io/downloads/edge) is installed on the edge node as an MQTT Broker.
```
Comment (fisherxu) [inline docs/setup/setup.md]: We can move this part below to `Run Edge Part/verify the configurations`.

**Comment by kevin-wangzefeng** [comment]
/lgtm
Again we can't preview rendered doc changes with files inside the main repo.

File: docs/setup/setup.md [hunk 2]
```diff
@@ -191,6 +194,17 @@ cd $GOPATH/src/github.com/kubeedge/kubeedge
...
+
```
Comment (fisherxu) [inline docs/setup/setup.md]: Could we use these lines replace the L307-308?

Comment (chendave) [reply docs/setup/setup.md]: > If you have run mosquitto on your edge host, please set mqtt.mode to 2

Is that meaning mqtt.mode must be set to 2 in this case?  If not, we can remove this statement. 

> To use KubeEdge in double mqtt or external mode (set mqtt.mode to 1), you need to make sure that mosquitto or emqx edge is installed on the edge node as an MQTT Broker.

This is duplicated, we can just remove that one.

Comment (fisherxu) [reply docs/setup/setup.md]: Yes, just delete them.

**Pull request merged**